### DISCUSSION TITLE
How to listen to theme change event?

### DISCUSSION THREAD

**Discussion by ifteakhor** [filamentphp/filament#12968] [Help] [answered]
### Package

Panel builder

### Package Version

v3.2.62

### How can we help you?

I need to set style of some elements base on theme mode, the styles work on first render but when user changes theme styles don't change. I need to list to change theme event, how can I do it?

My current js code:

```
        const isDark = Array.from(document.documentElement.classList).includes('dark');

        const userMessages = [...document.querySelectorAll('#user-message-box')];

        userMessages.forEach(message => {
            if (isDark) {
                message.style.background = '#075985'
            } else {
                message.style.background = '#72c9f8'
            }
        })
```

**Comment by leandrocfe** [comment, marked as answer]
I think you can use `theme-changed`

```js
window.addEventListener('theme-changed', (event) => {
    let theme = event.detail
})
```

**Reply by ifteakhor** [reply]
Thanks, it worked. A have an another question. How can I detect theme if user selects `system` ?
